crates/transaction-pool/src/validate.rs

```diff
@@ -80,6 +77,137 @@ pub trait TransactionValidator: Send + Sync {
+                    return Err(Error::TransactionEip2930Disabled)
+                }
+            }
+
+            2 => {
+                // Reject dynamic fee transactions until EIP-1559 activates.
+                if !self.eip1559 {
+                    return Err(Error::TransactionEip1559Disabled)
+                }
+            }
+
+            _ => return Err(Error::TxTypeNotSupported),
+        };
+
+        if self.shanghai {
+            match self.ensure_max_init_code_size(transaction.clone(), 2 * 24576) {
+                Ok(_) => {}
+                Err(e) => return Ok(TransactionValidationOutcome::Invalid(transaction, e)),
+            }
+        }
+
+        // Drop non-local transactions under our own minimal accepted gas price or tip
+        if !origin.is_local() && transaction.max_fee_per_gas() < self.gas_price {
+            return Err(Error::TransactionMaxFeeLessThenBaseFee)
+        }
+
+        // Ensure gasFeeCap is greater than or equal to gasTipCap.
+        if transaction.max_fee_per_gas() >= transaction.max_priority_fee_per_gas() {
+            return Err(Error::TransactionMaxFeeLessThenBaseFee)
+        }
+
+        // Checks for gas limit
+        if transaction.gas_limit() > self.current_max_gas_limit {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction,
+                PoolError::TxExceedsGasLimit(
+                    *invalid_transaction.hash(),
+                    invalid_transaction.gas_limit(),
+                    self.current_max_gas_limit,
+                ),
+            ))
+        }
+
+        // transaction size
+        if transaction.size() > 4 * 32 * 1024 {
+            return Ok(TransactionValidationOutcome::Invalid(
+                transaction,
+                PoolError::TxExceedsMaxInitCodeSize(
+                    *invalid_transaction.hash(),
+                    invalid_transaction.size(),
+                    4 * 32 * 1024,
+                ),
+            ))
+        }
+
+        // Checks for chainid
+        if transaction.chain_id() != self.chain_id {
+            return Err(Error::TransactionChainId)
+        }
+
+        // Checks for sender
+        let account = match self.client.basic_account(transaction.sender())? {
+            Some(account) => account,
+            None => {
+                return Ok(TransactionValidationOutcome::Invalid(
+                    transaction,
+                    PoolError::AccountNotFound(*invalid_transaction.hash()),
+                ))
+            }
+        };
+
+        // Checks for nonce
+        if transaction.nonce() < account.nonce {
+            return Err(Error::TransactionNonceNotConsistent)
+        }
+
+        // Checks for max cost
+        if transaction.cost() > account.balance {
+            return Err(Error::InsufficientFunds {
+                max_fee: transaction.max_fee_per_gas().unwrap_or_default(),
+                available_funds: account.balance,
+            })
+        }
```

Comment:
how does geth handle this case?
because this transaction may become valid in the future

Reply:
This is how [it does in geth](https://github.com/ethereum/go-ethereum/blob/5d8bff1d7261aed1a68235b0223d44a4fb4fb259/core/txpool/txpool.go#L642)